File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/features/users/activateUser/ActivateUserUseCase.java
package br.com.deltaglobalbank.identity.features.users.activateUser;

import java.util.UUID;

import br.com.deltaglobalbank.identity.domain.user.User;
import br.com.deltaglobalbank.identity.domain.user.UserNotFound;
import br.com.deltaglobalbank.identity.domain.user.UserRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class ActivateUserUseCase {

    private static final Logger log = LoggerFactory.getLogger(ActivateUserUseCase.class);

    private final UserRepository userRepository;

    public ActivateUserUseCase(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Transactional
    public void activateUser(UUID userId, UUID tenantId, UUID actorId) {
        User userExists = userRepository.findById(userId);
        if (userExists == null) {
            throw new UserNotFound();
        }
        if (!userExists.getTenantId().equals(tenantId)) {
            throw new UserNotFound();
        }
        userExists.activate();
        userRepository.save(userExists);
        log.info("user activated userId={} tenantId={} by={}", userId, tenantId, actorId);
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/features/users/activateUser/ActivateUserUseCase.kt
package br.com.deltaglobalbank.identity.features.users.activateUser

import br.com.deltaglobalbank.identity.domain.user.UserNotFound
import br.com.deltaglobalbank.identity.domain.user.UserRepository
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.UUID

@Service
class ActivateUserUseCase(private val userRepository: UserRepository) {
    private val log = LoggerFactory.getLogger(ActivateUserUseCase::class.java)

    @Transactional
    fun activateUser(userId: UUID, tenantId: UUID, actorId: UUID) {
        val userExists = userRepository.findById(userId) ?: throw UserNotFound()
        if(userExists.tenantId != tenantId) throw UserNotFound()
        userExists.activate()
        userRepository.save(userExists)
        log.info("user activated userId={} tenantId={} by={}", userId, tenantId, actorId)
    }
}